### ISSUE TITLE
Specify Workspace name

### ISSUE THREAD

**Issue by adamlc** [AzBuilder/terrakube-action-github#21]
I've just started to experiment and it looks like this action uses the folder name as the name of the workspace.

Would it be possible to specify the workspace so it doesn't rely on the folder name? Unless I'm missing something :)

**Comment by alfespa17** [comment]
Hello @adamlc 

It is using the workspace folder as the workspace name as you can see in this part of the code:

https://github.com/AzBuilder/terrakube-action-github/blob/9fd23568d70d51b62721d3181d01b213479cb1eb/src/main.ts#L44

https://github.com/AzBuilder/terrakube-action-github/blob/9fd23568d70d51b62721d3181d01b213479cb1eb/src/terrakube.ts#L55

It has been a long time since I update this action (1.5 years ago), maybe it will need a few adjustments.

**Comment by alfespa17** [comment]
There is one example that show how to use the action [here](https://github.com/AzBuilder/terraform-sample-repository)

**Comment by adamlc** [comment]
@alfespa17 yes thanks, I can confirm that using the folder name did indeed work. Ideally it would be great if we could just specify a workspace ID, otherwise we'll have to refactor quite a lot of our TF config :)

**Comment by alfespa17** [comment]
> @alfespa17 yes thanks, I can confirm that using the folder name did indeed work. Ideally it would be great if we could just specify a workspace ID, otherwise we'll have to refactor quite a lot of our TF config :)

I dont think this will require a lot of changes basically instead of searching the workspaceId [here](https://github.com/AzBuilder/terrakube-action-github/blob/9fd23568d70d51b62721d3181d01b213479cb1eb/src/main.ts#L44) using the workspaceFolder you could read the workspaceId from the "[terrakube.json](https://github.com/AzBuilder/terrakube-action-github/blob/9fd23568d70d51b62721d3181d01b213479cb1eb/src/main.ts#L20)" that for now only has the terraform version, reading data from that configuration file is quite simple like [this](https://github.com/AzBuilder/terrakube-action-github/blob/9fd23568d70d51b62721d3181d01b213479cb1eb/src/main.ts#L56C98-L56C121)

I dont have a lot of bandwith for the following weeks but if you would like to help with this feature all pull request are welcome
